src/Internal/Payment/PaymentContext.php

```diff
@@ -225,6 +225,53 @@ public function get_customer_id(): ?string {
 		return $this->get( 'customer_id' );
 	}
 
+	/**
+	 * Sets the previous paid duplicate order ID.
+	 *
+	 * @param  int $duplicate_order_id Duplicate order ID.
+	 *
+	 * @return void
+	 */
+	public function set_duplicate_order_id( int $duplicate_order_id ) {
+		$this->set( 'duplicate_order_id(', $duplicate_order_id );
+	}
+
+	/**
+	 * Gets the previous paid duplicate order ID.
+	 *
+	 * @return int|null
+	 */
+	public function get_duplicate_order_id(): ?int {
+		return $this->get( 'duplicate_order_id' );
+	}
+
+	/**
+	 * Sets the detected authorized intent ID.
+	 *
+	 * @param string $authorized_intent_id Payment intent ID.
+	 */
+	public function set_authorized_intent_id( string $authorized_intent_id ) {
+		$this->set( 'authorized_intent_id', $authorized_intent_id );
+	}
+
+	/**
+	 * Gets the detected authorized intent ID.
+	 *
+	 * @return string|null Payment intent ID.
+	 */
+	public function get_authorized_intent_id(): ?string {
```

Comment:
Agreed that we would use a boolean flag here. 
But it's not clear which `intent` property you mean here? 

But anway, since the intent ID is attached to the order already, we can retrieve it anytime.

Reply:
> But it's not clear which intent property you mean here?

Oh, I've just found out it https://github.com/Automattic/woocommerce-payments/pull/7471/files#diff-35fbbfe7a2e6ba1cab161794c92b9001d634b741162fb3f22106c3f03cfec83cR255-R266. This makes sens now.